test/distributed/org/apache/cassandra/distributed/test/OversizedMutationTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import org.apache.cassandra.distributed.Cluster;
+
+import static org.apache.cassandra.distributed.api.ConsistencyLevel.ALL;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class OversizedMutationTest extends TestBaseImpl
+{
+    @Test
+    public void testSingleOversizedMutation() throws Throwable
+    {
+        try (Cluster cluster = init(builder().withNodes(1).withConfig(c -> c.set("max_mutation_size_in_kb", 48))
+                                             .start()))
+        {
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.t (key int PRIMARY KEY, val blob)"));
+            String payload = StringUtils.repeat('1', 1024 * 49);
+            String query = "INSERT INTO %s.t (key, val) VALUES (1, textAsBlob('" + payload + "'))";
+            try
+            {
+                cluster.coordinator(1).execute(withKeyspace(query), ALL);
+                fail("An error should have been thrown but was not.");
+            }
+            catch (Throwable e)
+            {
+                String msg = "Rejected an oversized mutation (50233/49152) for keyspace: distributed_test_keyspace. Top keys are: t.1";
+                assertEquals(msg, e.getMessage());
+            }
```

Comment:
Nit: this can be slightly simplified using `assertThatThrownBy`:
```suggestion
            Assertions.assertThatThrownBy(() -> cluster.coordinator(1).execute(withKeyspace(query), ALL))
                      .hasMessage("Rejected an oversized mutation (50233/49152) for keyspace: distributed_test_keyspace. Top keys are: t.1");
```

Reply:
The actual size of `50233` might be a bit fragile if we change serialization formats, maybe we could omit it and do something like:
```java
Assertions.assertThatThrownBy(() -> cluster.coordinator(1).execute(withKeyspace(query), ALL))
          .hasMessageContaining("Rejected an oversized mutation (")
          .hasMessageContaining("49152) for keyspace: distributed_test_keyspace. Top keys are: t.1");
```
wdyt? The same would apply to the other test for batches below.

Reply:
Done in [5d7f657](https://github.com/apache/cassandra/pull/1588/commits/5d7f657cd1a3c8a5d841e99d1a3d670c0efddd53).